src/addressindex.h

```diff
@@ -11,6 +11,16 @@
 
 #include <chrono>
 
+namespace AddressType {
+enum AddressType {
```

Comment:
I have a commit that actually does use `enum class` and strictly enforces `AddressType` (https://github.com/kwvg/dash/commit/ac3a834031c4aac1c9f156d9b75b9b41a5093017) but I decided to leave it out. It could be brought back into this PR if there's consensus that it's a good fit.

Reply:
I'd prefer enum class as well

Reply:
Resolved in latest push.